Dockerfile

```diff
@@ -1,46 +1,20 @@
-# Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+# Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 FROM centos:7
 
 # Needed to build vespa
 RUN yum-config-manager --add-repo https://copr.fedorainfracloud.org/coprs/g/vespa/vespa/repo/epel-7/group_vespa-vespa-epel-7.repo && \
     yum -y install epel-release && \
     yum -y install centos-release-scl && \
-    yum -y --enablerepo=epel-testing install \
-        ccache \
-        cmake3 \
-        devtoolset-7-binutils \
-        devtoolset-7-gcc-c++ \
-        devtoolset-7-libatomic-devel \
-        flex \
-        bison \
-        git \
-        java-1.8.0-openjdk-devel \
-        Judy-devel \
-        libicu-devel \
-        libzstd-devel \
-        llvm3.9-devel \
-        llvm3.9-static \
-        lz4-devel \
-        make \
-        rh-maven35 \
-        openssl \
-        openssl-devel \
-        perl \
-        perl-Data-Dumper \
-        perl-Env \
-        perl-IO-Socket-IP \
-        perl-JSON \
-        perl-libwww-perl \
-        perl-Net-INET6Glue \
-        perl-URI \
-        rpm-build \
-        sudo \
-        valgrind \
-        'vespa-boost-devel >= 1.59.0-7' \
-        'vespa-cppunit-devel >= 1.12.1-7' \
-        'vespa-gtest >= 1.8.0-1' \
-        zlib-devel && \
-    yum clean all && \
+    yum -y --enablerepo=epel-testing install git yum-utils
+
+ENV GIT_REPO "https://github.com/vespa-engine/vespa.git"
+
+# Change git reference for a specific version of the vespa.spec file. Use a tag or SHA to allow for reproducible builds.
+ENV VESPA_SRC_REF "ac8b53179c27f71916fd4dc48852dad2f2e79764"
+
+RUN git clone $GIT_REPO && cd vespa && git -c advice.detachedHead=false checkout $VESPA_SRC_REF && \
```

Comment:
Using `--depth` on cloning can probably save some time.

Reply:
Setting the `--depth` argument is not a good idea as the required depth level will change as the ref gets older.

There are better approaches for optimizing the clone step (requiring newer git version): https://stackoverflow.com/a/30701724.